src/renderers/shared/fiber/ReactFiberUpdateQueue.js

```diff
@@ -367,14 +377,18 @@ exports.getPendingPriority = getPendingPriority;
 function addTopLevelUpdate(
   fiber : Fiber,
   partialState : PartialState<any, any>,
+  callback : Callback | null,
   priorityLevel : PriorityLevel
 ) : void {
-  const isTopLevelUnmount = partialState === null;
+  const isTopLevelUnmount = Boolean(
+    partialState &&
+    partialState.element === null
+  );
```

Comment:
At a glance, this change seems unrelated to the core of this PR, and also like the opposite of what the logic used to be. Was it a bug before? I don't see any other change inside of this PR that looks as though it would have invalidated the previous check, but maybe I'm missing it.